packages/roosterjs-content-model-core/lib/command/cutCopy/triggerBeforeCutCopyEvent.ts

```diff
@@ -0,0 +1,105 @@
+import { adjustImageSelectionOnSafari } from '../../corePlugin/copyPaste/utils/adjustImageSelectionOnSafari';
+import { adjustSelectionForCopyCut } from '../../corePlugin/copyPaste/utils/adjustSelectionForCopyCut';
+import { onCreateCopyEntityNode } from '../../override/pasteCopyBlockEntityParser';
+import { preprocessTable } from '../../corePlugin/copyPaste/utils/preprocessTable';
+import { pruneUnselectedModel } from '../../corePlugin/copyPaste/utils/pruneUnselectedModel';
+import type {
+    BeforeCutCopyEvent,
+    DOMSelection,
+    IEditor,
+    OnNodeCreated,
+} from 'roosterjs-content-model-types';
+import {
+    contentModelToDom,
+    createModelToDomContext,
+    isElementOfType,
+    isNodeOfType,
+    iterateSelections,
+    wrap,
+} from 'roosterjs-content-model-dom';
+
+/**
+ * @internal
+ * Exported only for unit testing
+ */
+export const onNodeCreated: OnNodeCreated = (modelElement, node): void => {
+    if (isNodeOfType(node, 'ELEMENT_NODE') && isElementOfType(node, 'table')) {
+        wrap(node.ownerDocument, node, 'div');
+    }
+    if (isNodeOfType(node, 'ELEMENT_NODE') && !node.isContentEditable) {
+        node.removeAttribute('contenteditable');
+    }
+    onCreateCopyEntityNode(modelElement, node);
+};
+
+/**
+ * Trigger the "beforeCutCopy" event
+ * @param editor The editor object
+ * @param isCut If the event to be created is cut. @default false
+ * @param event an optional parameter for when the clipboard event needs to be trigger
+ * @returns
+ */
+export function triggerBeforeCutCopyEvent(
```

Comment:
Since this function is mainly used for getting the copy/cut content, triggering event is just a step of it, let name it getContentForCopy.

Since cutting the content is not done in this function, so no need to mention "Cut" in the function name.